src/droplet.F

```diff
@@ -645,34 +764,48 @@ subroutine BE_integration(dt,np,pdata,pdata_locind,xf,yf,x3d,y3d, &
       
       guess = 0.0
 
+      t_f = mpi_wtime()
+      t_BEprep = t_BEprep + (t_f-t_s)
+
+      t_s = mpi_wtime()
+
       !Gives initial guess into nonlinear solver
       !mflag = 0, has equilibrium radius; mflag = 1, no equilibrium (uses itself as initial guess)
       call rad_solver2(guess,mflag,prsval,rhoval,tval,qval,pdata(np,prtp),pdata(np,prms),pdata(np,prrp),np)
 
       if (mflag.eq.0) then
-         rt_start(1) = guess/pdata(np,prrp)
-         rt_start(2) = tval/pdata(np,prtp)
+         rt_start(1) = guess/dble(pdata(np,prrp))
+         rt_start(2) = dble(tval)/dble(pdata(np,prtp))
       else
          rt_start(1) = 1.0
          rt_start(2) = 1.0
       end if
 
-      call gauss_newton_2d(diffnorm,dt_nondim,taup_scale,rt_start,rt_zeros,flag,prsval,rhoval,tval,qval,pdata(np,prtp),pdata(np,prms),pdata(np,prrp),np)
+      t_f = mpi_wtime()
+      t_radsolver = t_radsolver + (t_f-t_s)
 
+      t_s = mpi_wtime()
+      call gauss_newton_2d(dble(diffnorm),dble(dt_nondim),dble(taup_scale),rt_start,rt_zeros,flag,dble(prsval),dble(rhoval),dble(tval),dble(qval),dble(pdata(np,prtp)),dble(pdata(np,prms)),dble(pdata(np,prrp)),np)
+      t_f = mpi_wtime()
+      t_gauss = t_gauss + (t_f-t_s)
+
+      t_s = mpi_wtime()
       if (flag==1) then
          num100 = num100+1
-         call LV_solver(diffnorm,dt_nondim,taup_scale,rt_start,rt_zeros,flag,prsval,rhoval,tval,qval,pdata(np,prtp),pdata(np,prms),pdata(np,prrp),np)
+         call LV_solver(dble(diffnorm),dble(dt_nondim),dble(taup_scale),rt_start,rt_zeros,flag,dble(prsval),dble(rhoval),dble(tval),dble(qval),dble(pdata(np,prtp)),dble(pdata(np,prms)),dble(pdata(np,prrp)),np)
       end if
-
-      if (flag==1) num1000 = num1000+1
+      t_f = mpi_wtime()
+      t_LV = t_LV + (t_f-t_s)
 
       !In rare cases the nonlinear solvers fail to converge
-      if ( (rt_zeros(1) .ne. rt_zeros(1)) .OR. &
-           (rt_zeros(2) .ne. rt_zeros(2)) .OR. &
-           (rt_zeros(2) .lt. 0) ) then
-         !If this happens keep radius unchanged and temp equal to surrounding
-         rt_zeros(1) = 1.0
-         rt_zeros(2) = tval/pdata(np,prtp)
+      if (flag==1)  then
+
+       num1000 = num1000+1 
+
```

Comment:
The `if` checks for `rt_zeros` are removed, is it intended or not?

Reply:
Yes, this was intended. The checks on `rt_zeros` are now above, and they set `flag = 1` which triggers this code.